pkg/liqonet/tunnel/wireguard/driver.go

```diff
@@ -501,3 +511,45 @@ func (w *Wireguard) updateConnectionStatus(oldConn *netv1alpha1.Connection) (*ne
 		oldConn.PeerConfiguration[liqoconst.PublicKey], liqoconst.DeviceName)
 	return newConnectionOnError(err.Error()), err
 }
+
+// Collect implements prometheus.Collector.
+func (w *Wireguard) Collect(ch chan<- prometheus.Metric) {
+	device, err := w.client.Device(liqoconst.DeviceName)
+	w.MetricsCollectErrorHandler(err, ch)
```

Comment:
1. Should immediately return in case of errors.
2. Error messages shall be properly enriched with a descriptive message.